command/connect/envoy/envoy.go

```diff
@@ -0,0 +1,301 @@
+		useTLS = true
+	}
+
+	// We want to allow grpcAddr set as host:port with no scheme but if the host
+	// is an IP this will fail to parse as a URL with "parse 127.0.0.1:8500: first
+	// path segment in URL cannot contain colon". On the other hand we also
+	// support both http(s)://host:port and unix:///path/to/file.
+	addrPort := strings.TrimPrefix(c.grpcAddr, "http://")
+	addrPort = strings.TrimPrefix(c.grpcAddr, "https://")
+
+	agentAddr, agentPort, err := net.SplitHostPort(addrPort)
+	if err != nil {
+		return "", fmt.Errorf("Invalid Consul HTTP address: %s", err)
+	}
+	if agentAddr == "" {
+		agentAddr = "127.0.0.1"
+	}
+
+	// We use STATIC for agent which means we need to resolve DNS names like
+	// `localhost` ourselves. We could use STRICT_DNS or DYNAMIC_DNS with envoy
+	// but Envoy resolves `localhost` differently to go on macOS at least which
+	// causes paper cuts like default dev agent (which binds specifically to
+	// 127.0.0.1) isn't reachable since Envoy resolves localhost to `[::]` and
+	// can't connect.
+	agentIP, err := net.ResolveIPAddr("ip", agentAddr)
+	if err != nil {
+		return "", fmt.Errorf("Failed to resolve agent address: %s", err)
+	}
+
+	adminAddr, adminPort, err := net.SplitHostPort(c.adminBind)
+	if err != nil {
+		return "", fmt.Errorf("Invalid Consul HTTP address: %s", err)
+	}
+
+	// Envoy requires IP addresses to bind too when using static so resolve DNS or
+	// localhost here.
+	adminBindIP, err := net.ResolveIPAddr("ip", adminAddr)
+	if err != nil {
+		return "", fmt.Errorf("Failed to resolve admin bind address: %s", err)
+	}
+
+	args := templateArgs{
+		ProxyCluster:          c.proxyID,
+		ProxyID:               c.proxyID,
+		AgentHTTPAddress:      agentIP.String(),
+		AgentHTTPPort:         agentPort,
+		AgentTLS:              useTLS,
+		AgentCAFile:           httpCfg.TLSConfig.CAFile,
+		AdminBindAddress:      adminBindIP.String(),
+		AdminBindPort:         adminPort,
+		Token:                 httpCfg.Token,
+		LocalAgentClusterName: xds.LocalAgentClusterName,
+	}
+
+	var buf bytes.Buffer
+	err = t.Execute(&buf, args)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func (c *cmd) lookupProxyIDForSidecar() (string, error) {
+	return proxyCmd.LookupProxyIDForSidecar(c.client, c.sidecarFor)
+}
+
+func (c *cmd) Synopsis() string {
+	return synopsis
+}
+
+func (c *cmd) Help() string {
+	return c.help
+}
+
+const synopsis = "Runs or Configures Envoy as a Connect proxy"
+const help = `
+Usage: consul connect envoy [options]
+
+	Generates the bootstrap configuration needed to start an Envoy proxy instance 
+	for use as a Connect sidecar for a particular service instance. By default it 
+	will generate the config and then exec Envoy directly until it exits normally.
+
+	It will search $PATH for the envoy binary but this can be overridden with 
+	-envoy-binary.
+
+	It can instead only generate the bootstrap.yaml based on the current ENV and 
+	arguments using -bootstrap.
```

Comment:
Looks like there might be some mixed tab/non-tab usage here. I recommend copying the no-tab unit test from `commands/intention/create/create_test.go` to verify this is clean.